lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -430,12 +430,7 @@ export class UserAgentApplication {
         request.state
       );
 
-      // Cache nonce
-      this.cacheStorage.setItem(Constants.nonceIdToken, serverAuthenticationRequest.nonce, this.inCookie);
-
-      // Cache account and authority
-      this.setAccountCache(account, serverAuthenticationRequest.state);
-      this.setAuthorityCache(serverAuthenticationRequest.state, acquireTokenAuthority.CanonicalAuthority);
+      this.updateAcquireTokenCacheEntries(serverAuthenticationRequest, account);
```

Comment:
@pkanher617 As discussed you may want to combine the login calls caching code for "authority" and "nonce" also into this API. Please factor that login calls do not have "acquireTokenAccountKey".

Reply:
added to updateCacheEntries. Please check new code.